Detector/DetSensitive/src/SimpleCalorimeterSD.cpp

```diff
@@ -40,6 +41,36 @@ bool SimpleCalorimeterSD::ProcessHits(G4Step* aStep, G4TouchableHistory*)
   if(edep==0.)
     return false;
 
+  if(m_applyBirksLaw){
```

Comment:
This branch is processed on every hit and apart from Birk's law there is little code, so I thing a dedicated SD is a better solution.